src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -264,10 +280,13 @@ out _
 				}
 
 				diagnosticSink( diagnostic );
-				immutable = false;
+
+				// We're going to keep looking at all writes to surface as many
+				// relevant diagnostics as possible.
+				allAssignmentsAreOfImmutableValues = false;
 			}
 
-			return immutable;
+			return isReadOnly && allAssignmentsAreOfImmutableValues;

```

Comment:
I thought this was more readable anyway vs. reusing `immutable`: if we went looking for writes then in the end what we care about is that

1) our member is readonly
2) all the assignments are cool